docs/examples/201/shared-image-gallery-with-image-defintition-and-role-assignment/main.bicep

```diff
@@ -0,0 +1,69 @@
+param azureSubscriptionID string
+param sigName string
+param sigLocation string
+param imagePublisher string
+param imageDefinitionName string
+param imageOffer string
+param imageSKU string
+param imageLocation string
+param roleNameGalleryImage string
+param principalID string
+param templateImageResourceGroup string
+
+var assignableScopes = '/subscriptions/${azureSubscriptionID}/resourcegroups/${templateImageResourceGroup}'
+var imageDefinitionFullName = '${sigName}/${imageDefinitionName}'
+
+//Create Shard Image Gallery
+resource wvdsig 'Microsoft.Compute/galleries@2020-09-30' = {
+  name: sigName
+  location: sigLocation
+}
+
+//Create Image definitation
+resource wvdid 'Microsoft.Compute/galleries/images@2020-09-30' = {
+  name: imageDefinitionFullName
+  location: imageLocation
+  properties: {
+    osState: 'Generalized'
+    osType: 'Windows'
+    identifier: {
+      publisher: imagePublisher
+      offer: imageOffer
+      sku: imageSKU
+    }
+  }
+}
+
+//create role definition
+resource gallerydef 'Microsoft.Authorization/roleDefinitions@2018-01-01-preview' = {
+  name: guid(roleNameGalleryImage)
+  properties: {
+    roleName: roleNameGalleryImage
+    description: 'Custom role for network read'
+    permissions: [
+      {
+        actions: [
+          'Microsoft.Compute/galleries/read'
+          'Microsoft.Compute/galleries/images/read'
+          'Microsoft.Compute/galleries/images/versions/read'
+          'Microsoft.Compute/galleries/images/versions/write'
+          'Microsoft.Compute/images/write'
+          'Microsoft.Compute/images/read'
+          'Microsoft.Compute/images/delete'
+        ]
+      }
+    ]
+    assignableScopes: [
+      assignableScopes
+    ]
+  }
+}
+
+//create role assignment
+resource galleryass 'Microsoft.Authorization/roleAssignments@2020-04-01-preview' = {
+  name: guid(resourceGroup().id)

```

Comment:
This `guid()` function needs some more arguments. If I did another role assignment to the same rg with different properties and used only the rg id as an argument, I would get a conflict. I'd recommend adding the roleDefinition id and principal id:

```
guid(resourceGroup().id, gallerydef.id, principalId)
```

Reply:
Thanks, updated!